app/src/main/java/com/github/wnder/guessLocation/GuessFragment.java

```diff
@@ -0,0 +1,243 @@
+import com.github.wnder.networkService.NetworkService;
+import com.github.wnder.user.GlobalUser;
+import com.github.wnder.user.User;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.mapbox.geojson.GeoJson;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.maps.Style;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+
+import javax.inject.Inject;
+
+import dagger.hilt.android.AndroidEntryPoint;
+
+import static com.github.wnder.guessLocation.MapBoxHelper.addArrowToStyle;
+import static com.github.wnder.guessLocation.MapBoxHelper.addGuessToStyle;
+import static com.github.wnder.guessLocation.MapBoxHelper.addPictureToStyle;
+import static com.github.wnder.guessLocation.MapBoxHelper.drawCircle;
+import static com.github.wnder.guessLocation.MapBoxHelper.updateCircle;
+import static com.github.wnder.guessLocation.MapBoxHelper.zoomFromKilometers;
+
+@AndroidEntryPoint
+public class GuessFragment extends Fragment implements OnSeekBarChangeListener, OnMapReadyCallback {
+    //Different distances for the radius
+    private final int[] distances = {5, 10, 20, 50, 100, 500, 1000};
+
+    @Inject
+    public NetworkService networkInfo;
+
+    private TextView radiusTextView;
+    private MapView mapView;
+    private MapboxMap mapboxMap;
+    private LatLng cameraPosition;
+    private User user;
+
+    public GuessFragment() {}
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                              Bundle savedInstanceState){
+        Mapbox.getInstance(getActivity(), getString(R.string.mapbox_access_token));
+        View rootView = inflater.inflate(R.layout.fragment_seekbar, container, false);
+
+        user = GlobalUser.getUser();
+
+        mapView = rootView.findViewById(R.id.mapViewFragment);
+        mapView.onCreate(savedInstanceState);
+        mapView.getMapAsync(this);
+
+        double cameraLat = user.getPositionFromGPS((LocationManager) getActivity().getSystemService(Context.LOCATION_SERVICE), getActivity()).getLatitude();
+        double cameraLng = user.getPositionFromGPS((LocationManager) getActivity().getSystemService(Context.LOCATION_SERVICE), getActivity()).getLongitude();
+        cameraPosition = new LatLng(cameraLat, cameraLng);
+
+        SeekBar radiusSeekBar = rootView.findViewById(R.id.radiusSeekBar);
+        radiusTextView = rootView.findViewById(R.id.radiusTextView);
+        manageSeekBar(radiusSeekBar, radiusTextView);
+
+        FloatingActionButton guessButton = rootView.findViewById(R.id.navigationToGuessButton);
+        guessButton.setOnClickListener((view) -> openPreviewActivity());
+
+        return rootView;
+    }
+
+    private void setZoom(){
```

Comment:
You could have used the zoom class to have an animation on the zoom change. That being said it is more work and it work really well like that already so you don't need to change it for me to approve your PR.